python/neuroglancer/pipeline/task_queue.py

```diff
@@ -98,10 +161,20 @@ def insert(self, task):
             "tag": task.__class__.__name__
         }
 
-        self.api.insert(project=self._project,
-                        taskqueue=self._queue_name,
-                        body=body).execute(num_retries=6)
+        def cloud_insertion(api):
```

Comment:
Also, in case it wasn't clear, these function definitions are different from a class definition. every time `TaskQueue.insert` is called it instantiates a new function object. That's how it closes over `tid` and binds it.